### PR TITLE
Simplify cmake

### PR DESCRIPTION
**Pull request by jlhcrawford**
The examples and sample-kernels can now be built using hekit using `hekit build` and their respective recipe files `examples.toml` and `sample-kernels.toml`.

The CMake overall structure has had a rework and been simplified.

**Note:** The two new recipe files have lines `src_dir = ` which have hard coded paths to the HE Toolkit. They have been marked with FIXMEs and will require refactoring to be more flexible.

### CODE REVIEW

File: he-samples/cmake/gbenchmark.cmake [hunk 1]
```diff
@@ -3,6 +3,10 @@
 
 include(ExternalProject)
 
+if (NOT ${CMAKE_INSTALL_PREFIX})
```
Comment (hamishun) [inline he-samples/cmake/gbenchmark.cmake:6]: Comment why we need this please

File: he-samples/cmake/gflags.cmake [hunk 2]
```diff
@@ -3,6 +3,10 @@
 
 include(ExternalProject)
```
Comment (hamishun) [inline he-samples/cmake/gflags.cmake:5]: Comment why we need this

File: he-samples/cmake/gtest.cmake [hunk 3]
```diff
@@ -3,6 +3,10 @@
 
 include(ExternalProject)
 
+if (NOT ${CMAKE_INSTALL_PREFIX})
```
Comment (hamishun) [inline he-samples/cmake/gtest.cmake:6]: comment please

**Review by hamishun — CHANGES_REQUESTED** [review]

File: he-samples/cmake/gbenchmark.cmake (hunk 1 above)
Comment (jlhcrawford) [reply he-samples/cmake/gbenchmark.cmake:8]: Comment added.

File: he-samples/cmake/gflags.cmake (hunk 2 above)
Comment (jlhcrawford) [reply he-samples/cmake/gflags.cmake:5]: Comment added.

File: he-samples/cmake/gtest.cmake (hunk 3 above)
Comment (jlhcrawford) [reply he-samples/cmake/gtest.cmake:8]: Comment added.

**Review by hamishun — APPROVED** [review]

**Pull request merged**